### ISSUE TITLE
`ept_rvi_mode` and `hv_mode` always attemp to update on apply

### ISSUE THREAD

**Issue by ghost** [hashicorp/terraform-provider-vsphere#1902]
### Community Guidelines

- [X] I have read and agree to the [HashiCorp Community Guidelines ](https://www.hashicorp.com/community-guidelines).
- [X] Vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue initial description to help the maintainers prioritize.
- [X] Do not leave "+1" or other comments that do not add relevant information or questions.
- [X] If you are interested in working on this issue or have submitted a pull request, please leave a comment.

### Terraform

v1.4.6

### Terraform Provider

v2.3.1 and v2.4.0

### VMware vSphere

v8.0.1

### Description

After a virtual machine is created using the provider, after running an apply again Terraform wants to update the virtual-machine in-place as the fields `ept_rvi_mode` and `hv_mode` are constantly being changed even without touching those fields.
It looks like that the provider is using one of `automatic`, `on` or `off`, where VSphere unsets the field to an empty string `""`.

As a work-around I can use the following block
```
lifecycle {
    ignore_changes = [
      ept_rvi_mode,
      hv_mode
    ]
  }
```

### Affected Resources or Data Sources

resource/vsphere_virtual_machine

### Terraform Configuration

-

### Debug Output

-

### Expected Behavior

No changes

### Actual Behavior

```
# module.workers["worker-06"].vsphere_virtual_machine.main will be updated in-place
  ~ resource "vsphere_virtual_machine" "main" {
      + ept_rvi_mode                            = "automatic"
      + hv_mode﻿﻿                                 = "hvAuto"
         id                                     = "..."
        name                                    = "vm-01"
        tags                                    = []
        # (69 unchanged attributes hidden)
```

### Steps to Reproduce

Run a `terraform apply` immediatly followed by a new run of `terraform apply`

**Comment by uprightvinyl** [comment]
Seeing the same behaviour; ESXi & vCenter 8.0.1, Terraform v1.4.6, vSphere provider v2.4.0.

**Comment by tenthirtyam** [comment]
Both default to automatic. One option may be to set these to computed like was done for annotation recently but they would still use the validation. cc @appilon 

https://github.com/hashicorp/terraform-provider-vsphere/blob/29a4c6ce2e7723e3bbeb58c5f12d5292b82c1b6c/vsphere/virtual_machine_config_structure.go#L26-L36

https://github.com/hashicorp/terraform-provider-vsphere/blob/29a4c6ce2e7723e3bbeb58c5f12d5292b82c1b6c/vsphere/virtual_machine_config_structure.go#L127-L140

**Comment by captainfalcon23** [comment]
Same issue encountered today with latest provider and latest Vmware vCenter 8.0.1.

**Comment by gmintoco** [comment]
same issue for us with latest provider and vcenter 8

**Comment by kbpsec** [comment]
Same here. We have reproduced this on vCenter 8 and esxi 8. If vCenter 8 with esxi 7 works fine.

**Comment by pramodsharma62** [comment]
Any update on this? I am running into same issue with latest provider.

**Comment by DSI-BenThomas** [comment]
Can confirm also hit the issues on vCenter 8U1a and ESXi 8U1a (VCF5)

**Comment by LargoUsagi** [comment]
Can confirm this on hosts using ESXi and vCenter 8.0.2

**Comment by Hello-User** [comment]
Still an issue, even after manually defining the fields.

Latest version of the provider.
vCenter 8.0.1 (build 21815093) and ESXi 8.0.2 (build 22380479).

**Comment by smhick** [comment]
same issue for us with latest provider and vcenter 8

**Comment by micropbl4** [comment]
We face the same problem:
Terraform v1.6.0
Terraform Provider v2.5.1
VMware vSphere v8.0.1

**Comment by nc-td** [comment]
im hitting this too

**Comment by Aly-ona** [comment]
We hit same issue with:
Terraform v1.3.7
registry.terraform.io/hashicorp/vsphere v2.6.0
esxi: 8.0.2 22380479
vcenter: 8.0.2.00000

**Comment by smooti** [comment]
same issue

**Comment by Flumeded** [comment]
Same issue
Workaround for now to add

lifecycle {
    ignore_changes = [hv_mode, ept_rvi_mode]
}
  
  to the resource

**Comment by tenthirtyam** [comment]
@stoyan-hristov - can you look into this one as well?

**Comment by tenthirtyam** [comment]
Reverted in 2.8.1. Reopening.

**Comment by spacegospod** [comment]
Going to reintroduce the [fix](https://github.com/hashicorp/terraform-provider-vsphere/pull/2172) as a breaking change in the next major release

**Comment by DamienDaco** [comment]
Vsphere provider  2.8.1
Vsphere 8.0.2.00300

I'm still having the issue, VM are rebooting.
I've implemented the workaround with the lifecycle from above.

Just to make sure, is 2.8.1 supposed to fix the issue, or am I missing something?

**Comment by spacegospod** [comment]
> Vsphere provider 2.8.1 Vsphere 8.0.2.00300
> 
> I'm still having the issue, VM are rebooting. I've implemented the workaround with the lifecycle from above.
> 
> Just to make sure, is 2.8.1 supposed to fix the issue, or am I missing something?

Hey @DamienDaco 
The fix was released with 2.8.0 but it caused updates to running VMs created with older versions so we decided to issue a patch and revert this change in 2.8.1

We consider this fix to be a breaking change since it can interrupt running workloads because the update to the configuration triggers a power cycle on the VMs.

We try to allow breaking changes only on major releases, that is why the fix for this issue will likely land with v3.0.0

**Comment by tenthirtyam** [comment]
**Research Note**: Was reported in https://github.com/hashicorp/terraform-provider-vsphere/issues/2202#issuecomment-2124459375 that this issue is not observed in vSphere 7. Verification reequired.
